File: widgets/overlay.py
from PyQt5.QtCore import Qt
from PyQt5.QtWidgets import QLabel


class Overlay(QLabel):
    def __init__(self):
        super(Overlay, self).__init__()

        self.setAlignment(Qt.AlignCenter | Qt.AlignVCenter)
        self.setStyleSheet("background: rgba(0,0,0,80%); color: #CCCCCC; font-size: 200px;")

File: src/overlay.cpp
#include "overlay.h"

Overlay::Overlay(QWidget* parent)
  : QLabel(parent)
{
  setAlignment(Qt::AlignCenter | Qt::AlignVCenter);
  setStyleSheet("background: rgba(0,0,0,70%); color: #CCCCCC; font-size: "
                "200px; padding: 0px;");

  hide();
}

void
Overlay::show()
{
  resize(parentWidget()->size());
  QLabel::show();
  raise();
}
